route53_dmarc_app.tf

```diff
@@ -25,39 +48,20 @@ resource "aws_route53_record" "_amazonses_dmarc_TXT" {
 resource "aws_route53_record" "wildcard_report_dmarc_TXT" {
   provider = aws.route53resourcechange
 
-  name    = "*._report._dmarc.dmarc.${aws_route53_zone.cyber_dhs_gov.name}"
+  name    = "*._report._dmarc.${local.dmarc_domain_name}"
   records = ["v=DMARC1"]
   ttl     = 300
   type    = "TXT"
   zone_id = aws_route53_zone.cyber_dhs_gov.zone_id
 }
 
-resource "aws_route53_record" "dkim1_dmarc_CNAME" {
+resource "aws_route53_record" "dkim_dmarc_CNAME" {
   provider = aws.route53resourcechange
 
-  name    = "6na6lcj7onl5bco4ytfj4ud7p6t7kvtp._domainkey.dmarc.${aws_route53_zone.cyber_dhs_gov.name}"
-  records = ["6na6lcj7onl5bco4ytfj4ud7p6t7kvtp.dkim.amazonses.com"]
-  ttl     = 1800
-  type    = "CNAME"
-  zone_id = aws_route53_zone.cyber_dhs_gov.zone_id
-}
-
-resource "aws_route53_record" "dkim2_dmarc_CNAME" {
-  provider = aws.route53resourcechange
-
-  name    = "nsbndtrubsyckjqnb4wkv6xdkrqe3dk5._domainkey.dmarc.${aws_route53_zone.cyber_dhs_gov.name}"
-  records = ["nsbndtrubsyckjqnb4wkv6xdkrqe3dk5.dkim.amazonses.com"]
-  ttl     = 1800
-  type    = "CNAME"
-  zone_id = aws_route53_zone.cyber_dhs_gov.zone_id
-}
-
-resource "aws_route53_record" "dkim3_dmarc_CNAME" {
-  provider = aws.route53resourcechange
-
-  name    = "yhfkaco3ukhtbowt2bdvfz5czwuofitm._domainkey.dmarc.${aws_route53_zone.cyber_dhs_gov.name}"
-  records = ["yhfkaco3ukhtbowt2bdvfz5czwuofitm.dkim.amazonses.com"]
-  ttl     = 1800
+  count   = 3
+  name    = "${element(aws_ses_domain_dkim.dmarc_dkim.dkim_tokens, count.index)}._domainkey.${local.dmarc_domain_name}"
```

Comment:
* `3` is a magic number here.  Should it be `length(aws_ses_domain_dkim.dmarc_dkim.dkim_tokens)`?
* Is there a reason to use `element()` here instead of `aws_ses_domain_dkim.dmarc_dkim.dkim_tokens[count.index]`?
* We have been preferring `for_each` to `count` to avoid needless reprovisioning when element order changes.  For a list you can use `for_each: toset(list_name)`.  See, for example, [here](https://github.com/cisagov/cool-sharedservices-networking/blob/develop/transitgateway.tf#L69-L76) and [here](https://github.com/cisagov/cool-sharedservices-networking/blob/develop/transitgateway.tf#L47-L49).

Reply:
Just following the example.  I can change this tho:
https://www.terraform.io/docs/providers/aws/r/ses_domain_dkim.html

Reply:
The examples have not yet been updated to 0.12 syntax in many cases.